### PR TITLE
[TM-1198] Bulk Polygons delete

### PR DESCRIPTION
**Pull request by cesarLima1**
https://gfw.atlassian.net/browse/TM-1198

### CODE REVIEW

File: src/components/elements/Map-mapbox/utils.ts [hunk 1]
```diff
@@ -375,29 +372,82 @@ export const addSourceToLayer = (layer: any, map: mapboxgl.Map, polygonsData: Re
       map.removeSource(name);
     }
     const URL_GEOSERVER = `${GEOSERVER}/geoserver/gwc/service/wmts?REQUEST=GetTile&SERVICE=WMTS
-      &VERSION=1.0.0&LAYER=${WORKSPACE}:${name}&STYLE=&TILEMATRIX=EPSG:900913:{z}&TILEMATRIXSET=EPSG:900913&FORMAT=application/vnd.mapbox-vector-tile&TILECOL={x}&TILEROW={y}&RND=${Math.random()}`;
+      &VERSION=1.0.0&LAYER=${WORKSPACE}:${layerName}&STYLE=&TILEMATRIX=EPSG:900913:{z}&TILEMATRIXSET=EPSG:900913&FORMAT=application/vnd.mapbox-vector-tile&TILECOL={x}&TILEROW={y}&RND=${Math.random()}`;
```
Comment (pachonjcl) [inline src/components/elements/Map-mapbox/utils.ts:375]: Now there are many places where we are using `${GEOSERVER}/geoserver/gwc/service/wmts?REQUEST=GetTile&SERVICE=WMTS&VERSION=1.0.0&LAYER=${WORKSPACE}:${layerName}&STYLE=&TILEMATRIX=EPSG:900913:{z}&TILEMATRIXSET=EPSG:900913&FORMAT=application/vnd.mapbox-vector-tile&TILECOL={x}&TILEROW={y}&RND=${Math.random()}` where you only change the layer name part.

Create a function that will return given url and accept as the parameter only the layer name.
Update it everywhere.

File: src/components/elements/Map-mapbox/utils.ts [hunk 2]
```diff
@@ -366,7 +363,7 @@ export const addPopupToLayer = (
 };
 
 export const addSourceToLayer = (layer: any, map: mapboxgl.Map, polygonsData: Record<string, string[]> | undefined) => {
-  const { name, styles } = layer;
+  const { name, layerName, styles } = layer;
```
Comment (pachonjcl) [inline src/components/elements/Map-mapbox/utils.ts:366]: It is not clear what is the difference between `layerName` and `name` is.
Please use another name for this, for example mapboxgl uses `id`, if you want to have a different purpose for this, as an example the geoserver name, please use that as `geoserverLayerName` but right now it is not obvious for someone with a glance at the code what is the difference.

**Review by pachonjcl — CHANGES_REQUESTED** [review]

File: src/components/elements/Map-mapbox/MapControls/DeleteBulkPolygonsControl.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+import { ModalId } from "@/components/extensive/Modal/ModalConst";
+import ModalDeleteBulkPolygons from "@/components/extensive/Modal/ModalDeleteBulkPolygons";
+import { useMapAreaContext } from "@/context/mapArea.provider";
+import { useModalContext } from "@/context/modal.provider";
+import { useSitePolygonData } from "@/context/sitePolygon.provider";
+import { SitePolygon } from "@/generated/apiSchemas";
+
+import Button from "../../Button/Button";
+import Text from "../../Text/Text";
```
Comment (pachonjcl) [inline src/components/elements/Map-mapbox/MapControls/DeleteBulkPolygonsControl.tsx:11]: update the imports to not use relative imports

File: src/components/elements/Map-mapbox/utils.ts [hunk 4]
```diff
@@ -375,29 +372,82 @@ export const addSourceToLayer = (layer: any, map: mapboxgl.Map, polygonsData: Re
...
+export const addDeleteLayer = (layer: any, map: mapboxgl.Map, polygonsData: Record<string, string[]> | undefined) => {
+  const { name, layerName, styles } = layer;
+  if (map) {
+    if (map.getSource(name)) {
+      styles?.forEach((_: unknown, index: number) => {
+        map.removeLayer(`${name}-${index}`);
+      });
+      map.removeSource(name);
+    }
+    const URL_GEOSERVER = `${GEOSERVER}/geoserver/gwc/service/wmts?REQUEST=GetTile&SERVICE=WMTS
```
Comment (pachonjcl) [inline src/components/elements/Map-mapbox/utils.ts]: I guess coming back to the naming stuff, this is not the GEOSERVER_URL, it is about the tile, right?
can rename this?

**Review by pachonjcl — CHANGES_REQUESTED** [review]

File: src/components/elements/Map-mapbox/utils.ts [hunk 5]
```diff
@@ -362,29 +362,32 @@ export const addPopupToLayer = (
   }
 };
 
+const getGeojserverURL = (layerName: string) => {
```
Comment (pachonjcl) [inline src/components/elements/Map-mapbox/utils.ts:365]: typo

**Review by pachonjcl — CHANGES_REQUESTED** [review]

File: src/components/elements/Map-mapbox/utils.ts (hunk 1 above)
Comment (egrojMonroy) [reply src/components/elements/Map-mapbox/utils.ts]: function created

File: src/components/elements/Map-mapbox/utils.ts (hunk 2 above)
Comment (egrojMonroy) [reply src/components/elements/Map-mapbox/utils.ts]: name changed

File: src/components/elements/Map-mapbox/MapControls/DeleteBulkPolygonsControl.tsx (hunk 3 above)
Comment (egrojMonroy) [reply src/components/elements/Map-mapbox/MapControls/DeleteBulkPolygonsControl.tsx]: done

File: src/components/elements/Map-mapbox/utils.ts (hunk 4 above)
Comment (egrojMonroy) [reply src/components/elements/Map-mapbox/utils.ts]: changed to

File: src/components/elements/Map-mapbox/utils.ts (hunk 5 above)
Comment (egrojMonroy) [reply src/components/elements/Map-mapbox/utils.ts]: changed too

**Review by pachonjcl — APPROVED** [review]

**Pull request merged**